typedb-src/modules/ROOT/pages/typedb-in-20-queries.adoc

```diff
@@ -0,0 +1,725 @@
+//define (constraint for the value that can be inserted)
+
+We can limit possible values that an attribute can take by using a regular expression in the type definition.
+
+.Schema with regex example
+[,typeql]
+----
+define
+
+task sub entity,
+    owns status;
+status sub attribute, value string, regex "^(STARTED|STOPPED|DELETED)$";
+----
+
+The `status` attribute type has a value type of string, but its value is limited by a regular expression.
+That regular expression permits only the following exact variants of values: `STARTED`, `STOPPED`, or `DELETED`.
+
+If a query tries to create an instance of the `status` type with, for example, the value `Created`,
+it invokes the following error:
+
+.Error message
+----
+Error> [THW11] Invalid Thing Write: Attempted to put an instance of 'status' with value 'Created' that does not satisfy the regular expression '^(STARTED|STOPPED|DELETED)$'.
+----
+
+//For more information on how to use regular expressions to limit attribute values,
+//see the xref:typeql::data/advanced.adoc[] section.
+
+== #{counter:query}: Match data with regex
+//match (constraint for the valued matched)
+
+Use the `like` keyword with a regular expression to set constraints for an attribute value in a `match` clause.
+
+.Query with regex example
+[,typeql]
+----
+match
+    $f isa file, has path $p;
+    $p like "^docs/.*$";
+----
+
+The query above will find all files with path starting with `docs/`.
+
+== #{counter:query}: Use arithmetic expressions
+_Run this query after <<_schema,query #1>> and <<_data,query #2>>._
+
+Arithmetic expressions let you perform basic math operations on any value in a `match` clause.
+
+See the list of available xref:typedb::fundamentals/patterns.adoc#_arithmetic_operations[arithmetic operations].
+
+Let's insert a few files with different sizes, that are calculated:
+
+.Insert files with file sizes
+[,typeql]
+----
+match
+    $f isa file, has path "docs/quickstart-guide.adoc", has size-kb $s;
+    ?x = ($s * 2) + 200;
+    ?y = abs ((?x * 3) - 1);
+insert
+    $f1 isa file, has path "config.yaml", has size-kb ?x;
+    $f2 isa file, has path "logs.zip", has size-kb ?y;
+----
+
+The above query inserts two files with `size-kb` values calculated based on the value of the `size-kb`
+of the matched existing file.
+See the resulted data visualization on the image below.
+
+image::typedb-in-20-queries/math.png[Arithmetics, role=framed, width = 75%, link=self]
+
+[#_rule_based_inference]
+== #{counter:query}: Discover rule-based inference
+_Run this query after <<_schema,query #1>> and <<_data,query #2>>._
+
+TypeDB reasoning engine can perform rule-based inference when you read data.
+To use inference, we need to xref:typeql::schema/define-types.adoc[Define] rules in the schema,
+open a `read` transaction with the `infer` option xref:typedb::development/infer.adoc#_how_to_use_inference[enabled],
+and send a xref:typeql::data/get.adoc[Get] query.
+
+.Schema with a rule
+[,typeql]
+----
```

Comment:
This example could be made a bit more interesting (even though admittedly it is funny).